### DISCUSSION TITLE
Next/Image component does not work with blob URL

### DISCUSSION THREAD

**Discussion by ghost** [vercel/next.js#19732] [Help]
I have a issue with Next/Image in Next ver 10.0.3
I am creating a upload image feature, previous in Next ver 9, it's work perfectly. But after update to new version of Next, it's does not work, and I get a error.
Although `localhost` was add to `next.config.js`, it's still not working
And my question is, How to fix this ? 🙇 

![image](https://user-images.githubusercontent.com/53286853/100852002-7d8dbd80-34b8-11eb-867f-b3ee40b8ee84.png)
![image](https://user-images.githubusercontent.com/53286853/100852067-8da59d00-34b8-11eb-97a4-762d09ff42bf.png)
![image](https://user-images.githubusercontent.com/53286853/100840298-41eaf780-34a8-11eb-8ea0-cb3e5b99ac9a.png)

**Comment by Tomburgs** [comment]
It's true that `next/image` currently does not support blob urls, #19291 is an active issue that has been added to the iteration 14 milestone.

A hacky solution to overcome this error is to add an empty string as domain to `next.config.js`, like so:
```js
...
images: {
  domains: ['']
}
...
```

I'm not sure how effective this solution is, as I haven't done much testing with it.
Also do note that this will allow blob urls from all locations, not just the current domain.

**Reply by ghost** [reply]
My solution is very easy, cause `Next/Image` do not support for `blob` [(here)](next.js/packages/next/client/image.tsx) so i don't use `blob`, [this](https://stackoverflow.com/a/4459419) is best practice i founded

**Comment by lordvcs** [comment]
for some reason using the loader function to return the `src` seems to make it work along with the empty quotes for the domains array

**Comment by marcin-piechaczek** [comment]
Any update on this issue?

**Reply by hilmanski** [reply]
check my answer, let's see if it does work for you

**Comment by Debarshi95** [comment]
Facing the same issue. Any update?

**Reply by hilmanski** [reply]
check my answer, let's see if it does work for you

**Reply by Debarshi95** [reply]
Thanks it's working now. Seems like some kind of hack..

**Reply by hilmanski** [reply]
that's how we solve every problem 😂 , a lot of hack

**Comment by hilmanski** [comment]
Hi, my current solution for this is to create a custom loader
```
  const customImgLoader = ({ src }) => {
        return `${src}`
    }

//later to use
<Image loader={customImgLoader} alt="Avatar" width=64 height=64 src={avatarUrl} />
```

We tell the next/image to return src as it is

[Update]
check the answer from @eashish93 
we can use unoptimized=false as attribute

**Reply by wahablukman** [reply]
Thanks @hilmanski works great !

**Reply by FahmyChaabane** [reply]
@hilmanski  I love you

**Reply by Dominique-Sakrisson** [reply]
@hilmanski this still works great!

**Reply by great084** [reply]
great Thanks

**Comment by eashish93** [comment]
Simply use `unoptimized` prop, if using blob url.
See: https://nextjs.org/docs/api-reference/next/image#unoptimized

**Reply by Chukstart** [reply]
Though when one wants to user an optimized blob image also commenting to get an update when it's supported

**Reply by eashish93** [reply]
Blob urls resides in memory, so it's temporary. Therefore not required for optimisation, as we have to clear it after page navigation or before closing the tab.

**Reply by iefendiev** [reply]
thank you very much!!!

**Reply by 0x7An** [reply]
Not all heroes wear capes. Thank you @eashish93

**Comment by stefanprobst** [comment]
fyi, blob urls should now be handled correctly in version `11.1.1-canary.7`.

**Reply by bruceharrison1984** [reply]
What do you mean by handled correctly? Using `12.0.7` requires me to still use the `unoptimized` flag. Is that the correct behavior, or should I be able to use it without the flag?

**Comment by gtrabanco** [comment]
Version 12.5 still fails... but works great with <img> html element

**Comment by itaquito** [comment]
Currently, I'm using version `13.1.5` and it works as intended, without needing the `unoptimized` flag or the empty string in `next.config.js`.

**Comment by KAEN7** [comment]
Do you still have this problem? It is not working in Next `13.4.12`.

```
const customImgLoader = ({ src }: { src: string }) => {
      return `${src}`;
};

...
   
<Image
      loader={customImgLoader}
      src={src}
      alt={blog?.title}
      fill
      unoptimized={false}
  />
```

**Reply by Deathstar10** [reply]
Hey did you get a solution for this in Next js 13.4.12 ? .

The alternative solution would be to  upload the file to cloud and then fetch the URL from it?

**Comment by Mitterdini** [comment]
has this been resolved? I have an image response that can't be rendered as a blob using next/image. the unoptimized method is not working for me.

**Comment by joetechster** [comment]
You can now use wild cards to accept all hostnames, this works well with Blobs.

Add this to your nextConfig object in your project's next.config.js file .
`images: {
        remotePatterns: [{ hostname: "*" }],
}`
**_Note: Single * matches a single subdomain. Double ** matches any number of subdomains._**
